src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs

```diff
@@ -46,27 +57,13 @@ public static IEnumerable<string> GetAllRuntimeIdentifiers(this IRuntimeEnvironm
             else
             {
                 var arch = env.RuntimeArchitecture.ToLowerInvariant();
-                if (env.OperatingSystemVersion.Equals("7.0", StringComparison.Ordinal))
-                {
-                    yield return "win7-" + arch;
-                }
-                else if (env.OperatingSystemVersion.Equals("8.0", StringComparison.Ordinal))
-                {
-                    yield return "win8-" + arch;
-                    yield return "win7-" + arch;
-                }
-                else if (env.OperatingSystemVersion.Equals("8.1", StringComparison.Ordinal))
-                {
-                    yield return "win81-" + arch;
-                    yield return "win8-" + arch;
-                    yield return "win7-" + arch;
-                }
-                else if (env.OperatingSystemVersion.Equals("10.0", StringComparison.Ordinal))
+                Version parsedVersion;
+                if (!Version.TryParse(env.OperatingSystemVersion, out parsedVersion))
+                    yield break;
+
+                foreach (var version in _windowsVersionMapping.Where(p=>p.Key <= parsedVersion).Select(p=>p.Value))
```

Comment:
I think the loop could be just: 
`return _windowsVersionMapping.Where(p => p.Key <= parsedVersion).Select(p => $"win{ToShortVersion(version)}-{arch}")`;

However AFAIR there was a concern of using LINQ in the lower layers of the host due to performance hit.. - @anurse